api/document/js/serialize-doc.ts

```diff
@@ -99,10 +115,26 @@ export function nestMarks(text: string, marks: Mark[]): ApiContent {
   return converted;
 }
 
+function convertInlineFootnote(node: Node): ApiContent {
+  const emblem = node.attrs.emblem;
+  return apiFactory.content('footnote_citation', {
+    inlines: [apiFactory.text(emblem)],
+    footnote_node: apiFactory.node('footnote', {
```

Comment:
This field isn't writable, so we can probably skip it here for clarity.

Reply:
Oh, so I'm actually "stashing" it there so I can retrieve it later in `extractFootnotes`.  I actually did this for a few reasons:

1. Maybe this field _should_ be writable?  It's clearly easier for this client code to write to it, so maybe it's just a better API.  In other words, maybe `extractFootnotes` should really be on the server-side. If we end up doing this, fixing this code would just involve getting rid of `extractFootnotes`, rather than re-architecting anything.

2. Because the functions in `serialize-doc.ts` return `ApiContent`, there's otherwise no easy way to return an `ApiNode` along with the content, so it's convenient to "stash" the footnote there. An alternative would be to modify the module to use functions that returned some sort of structure that included both types of content, e.g. the [`Uninlineds` structure from PR 1018](https://github.com/18F/omb-eregs/pull/1018/files#diff-7dbfa13a162e6c340db15585111acc4bR59).

Reply:
Ah, cool, I did not follow that this was being used again on first read-through. I suspect we'll change the format pretty significantly to match AKN: removing distinct footnote nodes and instead including all their text in the paragraph, along with a footnote _annotation_. In that case, the API'd be something like
```js
{
  "content_type": "footnote",
  "inlines": [{"content_type": "__text__", "text": "some content here"}],
  "number": "12",
}
```